crates/transaction-emitter-lib/src/transaction_generator/call_custom_modules.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    publishing::{module_simple::EntryPoints, publish_util::Package},
+    TransactionExecutor,
+};
+use crate::transaction_generator::{
+    publishing::publish_util::PackageHandler, TransactionGenerator, TransactionGeneratorCreator,
+};
+use aptos_infallible::RwLock;
+use aptos_logger::{info, sample, sample::SampleRate, warn};
+use aptos_sdk::{
+    transaction_builder::TransactionFactory,
+    types::{transaction::SignedTransaction, LocalAccount},
+};
+use async_trait::async_trait;
+use rand::{prelude::StdRng, seq::SliceRandom, Rng};
+use rand_core::{OsRng, SeedableRng};
+use std::{sync::Arc, time::Duration};
+
+pub struct CallCustomModulesGenerator {
+    rng: StdRng,
+    txn_factory: TransactionFactory,
+    packages: Arc<Vec<Package>>,
+    accounts_pool: Option<Arc<RwLock<Vec<LocalAccount>>>>,
+    entry_point: EntryPoints,
+}
+
+impl CallCustomModulesGenerator {
+    pub fn new(
+        rng: StdRng,
+        txn_factory: TransactionFactory,
+        packages: Arc<Vec<Package>>,
+        accounts_pool: Option<Arc<RwLock<Vec<LocalAccount>>>>,
+        entry_point: EntryPoints,
+    ) -> Self {
+        Self {
+            rng,
+            txn_factory,
+            packages,
+            accounts_pool,
+            entry_point,
+        }
+    }
+}
+
+#[async_trait]
+impl TransactionGenerator for CallCustomModulesGenerator {
+    fn generate_transactions(
+        &mut self,
+        accounts: Vec<&mut LocalAccount>,
+        transactions_per_account: usize,
+    ) -> Vec<SignedTransaction> {
+        let needed = accounts.len() * transactions_per_account;
+        let mut requests = Vec::with_capacity(needed);
+
+        let mut accounts_to_burn = if let Some(accounts_pool_lock) = &self.accounts_pool {
+            let mut accounts_pool = accounts_pool_lock.write();
+            let num_in_pool = accounts_pool.len();
+            if num_in_pool < needed {
+                sample!(
+                    SampleRate::Duration(Duration::from_secs(10)),
```

Comment:
Can you explain when will this happen? This doesn't look like a recoverable error to me?

Reply:
- if we run both transaction types, with mixed weights, simultaneously, then the pool is both inserted and taken from at the same time, so it can get replenished
- if we run sequentially, it isn't recoverable - but better to skip creating txns, then to panic (i.e. if there is next phase afterwards, etc)